src/lib/transition_frontier_controller_tests/test_sync_handler.ml

```diff
@@ -0,0 +1,59 @@
+open Core
+open Async
+open Coda_base
+open Stubs
+
+let%test_module "Sync_handler" =
+  ( module struct
+    let%test "sync with ledgers with another peer via glue_sync_ledger" =
+      Backtrace.elide := false ;
+      Printexc.record_backtrace true ;
+      let logger = Logger.create () in
+      let num_breadcrumb_to_add = 3 in
+      let max_length = num_breadcrumb_to_add + 2 in
+      Thread_safe.block_on_async_exn (fun () ->
+          let%bind frontier = create_root_frontier ~max_length ~logger in
+          let%bind () =
+            build_frontier_randomly frontier
+              ~gen_root_breadcrumb_builder:(fun root_breadcrumb ->
+                Quickcheck.Generator.with_size ~size:num_breadcrumb_to_add
+                @@ Quickcheck_lib.gen_imperative_list
+                     (root_breadcrumb |> return |> Quickcheck.Generator.return)
+                     (gen_breadcrumb ~logger) )
+          in
+          let source_ledger =
+            Transition_frontier.root frontier
+            |> Transition_frontier.Breadcrumb.staged_ledger
+            |> Staged_ledger.ledger
+          in
+          let dest_ledger =
+            Transition_frontier.best_tip frontier
+            |> Transition_frontier.Breadcrumb.staged_ledger
+            |> Staged_ledger.ledger
```

Comment:
isn't it a huge violation of the transition frontier's invariants to directly mutate any ledger stored in a breadcrumb? I'd prefer if this test wrote into an ephemeral mask or something.